DAOS-14010 rebuild: incremental reintegration

wangdi1 opened the pull request:
### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium MD on SSD completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13358/3/execution/node/1467/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-13358/3/testReport/

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13358/3/execution/node/1513/log

daltonbohning reviewed (changes requested):
This needs to run with
`Features: rebuild`
At least the modified test `src/tests/ftest/control/log_entry.py` is not included in `pr` and other rebuild tests are likely affected by these changes

wangdi1 committed:
DAOS-14010 placement: refine for delay rebuild

Fix a few typo for delay rebuild.

Required-githooks: true

Signed-off-by: Di Wang <di.wang@intel.com>

wangdi1 committed:
DAOS-14010 rebuild: incremental reintegration

Refine container and pool layout upgrade process.

Track minimum stable epoch from active and commit DTX
list.

Reintegrate from stable epoch.

Required-githooks: true
Features: rebuild
Allow-test-unstable: true
Signed-off-by: Di Wang <di.wang@intel.com>

daltonbohning commented on src/tests/ftest/daos_test/suite.yaml:
```diff
@@ -22,7 +22,7 @@ timeouts:
   test_daos_epoch_recovery: 104
   test_daos_md_replication: 104
-  test_daos_rebuild_simple: 1800
+  test_daos_rebuild_simple: 3600
```
This is an extra 30 minutes for a PR test. Can this instead be broken into multiple test cases where one runs in PR and another in daily or weekly?

mchaarawi commented:
@jolivier23 could you comment whether this PR is still needed or can be closed?
